client/src/app/shared/user-session/state/user-session.service.ts

```diff
@@ -0,0 +1,33 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { EnvService } from '@shared/env.service';
+import { ToastrService } from 'ngx-toastr';
+import { UserSession } from './user-session.model';
+import { UserSessionStore } from './user-session.store';
+
+@Injectable({ providedIn: 'root' })
+export class UserSessionService {
+  readonly apiRootUrl: string;
+
+  constructor(
+    envService: EnvService,
+    private http: HttpClient,
+    private store: UserSessionStore,
+    private toastr: ToastrService,
+  ) {
+    this.apiRootUrl = envService.apiRootUrl;
+  }
+
+  getUserSession(token: string) {
```

Comment:
I think the commented out section below is actually what you want. This method should return `: void`